ethereum-consensus/src/networks.rs

```diff
@@ -1,5 +1,43 @@
 /// This module contains support for various Ethereum netowrks.
-use crate::state_transition::Context;
+use crate::state_transition::{Context, Error};
+
+#[derive(Default, Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(rename_all = "lowercase"))]
+pub enum Network {
+    #[default]
+    Mainnet,
+    Sepolia,
+    Goerli,
+    Holesky,
+    Custom(String),
+}
+
+impl std::fmt::Display for Network {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Mainnet => write!(f, "mainnet"),
+            Self::Sepolia => write!(f, "sepolia"),
+            Self::Goerli => write!(f, "goerli"),
+            Self::Holesky => write!(f, "holesky"),
+            Self::Custom(config) => write!(f, "custom network with config at `{config}`"),
```

Comment:
this was written for a config file, let's handle that specific concern elsewhere and for now just keep `Custom` as a variant to wrap some custom network name
```suggestion
            Self::Custom(name) => write!(f, "{name}"),
```